modin/backends/pandas/parsers.py

```diff
@@ -102,12 +102,8 @@ def parse(fname, **kwargs):
             bio = FileDispatcher.file_open(
                 fname, "rb", kwargs.pop("compression", "infer")
             )
-            if kwargs.get("encoding", None) is not None:
-                header = b"" + bio.readline()
-            else:
-                header = b""
             bio.seek(start)
-            to_read = header + bio.read(end - start)
```

Comment:
Are you sure that the rest of the dataframe will be parsed in the same way?(without header)

Reply:
Looks like header is skipped anyway if `encoding` is not None because of this [code](https://github.com/modin-project/modin/pull/2607/files#diff-172c7ff4cc90a4dd253fa690e0165e9ae1d8a79214bdde6a1def95546be7cb95L129-L130)
Actually i can't find a scenarios when this lines should be used

Reply:
Yes, parser changes were rollbacked.